controllers/defaults.go

```diff
@@ -0,0 +1,31 @@
+package controllers
+
+// Below are the default constants that we will
+// use throughout the patterns operator code
+const (
+	// Default Operator Namespace
+	OperatorNamespace = "openshift-operators"
+	// Default Operator Config Map File Name
+	OperatorConfigFile = "patterns-operator-config"
+)
+
+// GitOps Subscription
+const (
+	GitOpsDefaultChannel                = "gitops-1.8"
+	GitOpsDefaultPackageName            = "openshift-gitops-operator"
+	GitOpsDefaultCatalogSource          = "redhat-operators"
+	GitOpsDefaultCatalogSourceNamespace = "openshift-marketplace"
+	GitOpsDefaultApprovalPlan           = "Automatic"
+)
```

Comment:
Here and in the places referencing things you could simplify things by creating a new type like:
```
var DefaultGitOpsConfig = map[string]string {
    "gitops.catalogSource": "gitops-1.8",
    "gitops.sourceNamespace": "openshift-gitops-operator",
    "gitops.name": "redhat-operators",
    "gitops.channel": "openshift-marketplace",
    "gitops.csv":  "Automatic",
}

type GitOpsConfig map[string]string

func (g GitOpsConfig)getValueWithDefault(k string) (string, error) {
    if v, present := g[k]; present {
                return v, nil
        }
    if defaultValue, present := DefaultGitOpsConfig[k]; present {
        return defaultValue, nil
    }
    return "", fmt.Errorf("Could not find neither an existing value nor a default for %s", k)
}
```

Maybe the defaultgitopsconfig can live here in defaults.go and the new type + func either in a separate file or in utils.go? wdyt?

Reply:
Maybe we can call it DefaultPatternOperatorConfig actually, as that seems more in line with the rest

Reply:
This is similar to what was implemented in utils.go.  Removed the utils.go impl and replaced it with this one.  Only difference is that it will only return a string.  If the map key is not there it will return an empty string.